GetStream/stream-chat-react-native
# feat: support automatically translated messages

Our backend supports automatically translated messages - this PR adds support for this in our SDK.

This fixes #1255 

## 🛠 Implementation details

Similarly to the React SDK, this implementation uses the current user language as set in the client (`client.user?.language?`), and will display messages translated to this language if one exists in `message.i18n?[${userLanguage}_text]`.

Where the text is rendered, `message.text` is still used - I figured it was neater to create a new message object with the correct translation set as the text and pass that to the components rendering the text, rather than explicitly looking up the translation in the places it's rendered.

The affected places text is rendered:

- Messages (MessageTextContainer)
- Channel Preview
- Message Overlay

## 🎨 UI Changes

No UI changes are relevant for this

## 🧪 Testing

1. Enable automated translations for the (Stream) app
2. Log in with two different users with different languages set (set `language` in the user object when connecting the user)
3. Send messages from each of the users to eachother in the language they have set 
4. See that the messages are being translated for the recipient

## ☑️ Checklist
